src/libraries/System.Text.Json/tests/Serialization/ExtensionDataTests.cs

```diff
@@ -229,6 +236,7 @@ public static void InvalidExtensionPropertyFail()
 
             Assert.Throws<InvalidOperationException>(() => JsonSerializer.Deserialize<ClassWithInvalidExtensionProperty>(@"{}"));
             Assert.Throws<InvalidOperationException>(() => JsonSerializer.Deserialize<ClassWithTwoExtensionProperties>(@"{}"));
+            Assert.Throws<NotSupportedException>(() => JsonSerializer.Deserialize<ClassWithNotSupportedExtensionPropertyImmutableDictionary>(@"{}"));
```

Comment:
Per my comment above, deserializing this empty payload should work fine.

Reply:
Can you add serialization tests as well? When the dictionary is `null`, empty, and has data?

Reply:
Can do.

Reply:
fixed in 388a28f